dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPIImpl.java

```diff
@@ -67,12 +73,41 @@ public Experiment save(final Experiment experiment, final User user) throws
 
         builder.modDate(Instant.now());
         builder.lastModifiedBy(user.getUserId());
+        
+        if(experiment.goals().isPresent()) {
+            validateGoals(experiment.goals().get());
+        }
 
         final Experiment experimentToSave = builder.build();
 
         return factory.save(experimentToSave);
     }
 
+    private void validateGoals(Goals goals) {
+        final Metric primaryGoal = goals.primary();
+
+        final Set<String> availableParams = primaryGoal.type()
+                .availableParameters().stream().map(Parameter::name).collect(Collectors.toSet());
+
+        final Set<String> providedParams = primaryGoal.conditions()
+                .stream().map(Condition::parameter).collect(Collectors.toSet());
+
+        if(!availableParams.containsAll(providedParams)) {
+            providedParams.removeAll(availableParams);
```

Comment:
I think we don't need to remove here

Reply:
We need remove because I want to include in the error message the invalid parameters only, which is `provided minus available`